lib/parser.rb

```diff
@@ -6,11 +8,21 @@ def initialize(file_pathname)
   end
 
   def get_first_line
-    if File.file?(file_pathname)
-      first_line = File.foreach(file_pathname).first
-    else
-      raise 'file or directory not found'
-    end
+    first_line = File.foreach(file_pathname).first unless validate_file
```

Comment:
Você checou na documentação se o foreach irá fechar o arquivo aberto em memória após a execução?

Reply:
```suggestion
    File.foreach(file_pathname).first unless validate_file
```

Não precisa declarar em uma variável se não for usar